snyk/cli
# Change `iac gen-driftignore` to `iac update-exclude-policy`

#### What does this PR do?

This PR move the responsibility to generate drift exclusion rule from driftctl to snyk CLI.
We are now generating our own rules in the snyk policy file instead of using driftctl to generate a `.driftignore`
More context about that choice in the Jira ticket.


#### Where should the reviewer start?

`src/cli/commands/update-exclude-policy.ts`

#### How should this be manually tested?

```shell
$ cat test/fixtures/iac/drift/analysis.json | bin/snyk iac update-exclude-policy
```

Ensure your .snyk file is update with those new exclude rules

```diff
+exclude:
+  iac-drift:
+    - aws_iam_access_key.AKIA5QYBVVD25KFXJHYJ
+    - aws_iam_user.test-driftctl2
+    - aws_iam_access_key.AKIA5QYBVVD2Y6PBAAPY
+    - aws_s3_bucket_policy.driftctl
+    - aws_s3_bucket_notification.driftctl

```

#### Any background context you want to provide?

The exclude logic is based on [this code](https://github.com/snyk/driftctl/blob/main/pkg/analyser/analysis.go#L259-L291) from driftctl

Changes in help files are just FYI here, changes are in a draft PR in gitbook

#### What are the relevant tickets?

https://snyksec.atlassian.net/browse/CFG-1669

#### Screenshots


#### Additional questions
